tdms/src/simulation_parameters.cpp

```diff
@@ -70,7 +73,9 @@ void SimulationParameters::set_Np(FrequencyExtractVector &f_ex_vec) {
 void SimulationParameters::unpack_from_input_matrices(
         InputMatrices in_matrices) {
   // determine if we have a dispersive medium or multilayer
-  CCollection C(in_matrices["C"]);
+  CCollection C;
+  HDF5Reader INPUT_FILE(in_matrices.input_filename);
+  INPUT_FILE.read(C);
```

Comment:
Why are you shouting? `INPUT_FILE` is not global, right?

Reply:
No, but this is currently an ugly bit of code - creating multiple instances of `HDF5Reader` to setup `SimulationManager` in each of `SimulationParameters`, `OutputMatrices` and `InputMatrices`.

Ideally `INPUT_FILE` would be passed from SimulationManager`.